core/primitives/src/runtime/parameter_table.rs

```diff
@@ -67,8 +69,34 @@ impl ParameterTable {
         })
     }
 
-    pub(crate) fn apply_diff(&mut self, diff: ParameterTable) {
-        self.params.extend(diff.params)
+    pub(crate) fn apply_diff(
+        &mut self,
+        diff: ParameterTableDiff,
+    ) -> Result<(), InvalidConfigError> {
+        for (key, (before, after)) in diff.params {
+            if before.is_null() {
+                match self.params.get(&key) {
+                    Some(serde_json::Value::Null) | None => {
+                        self.params.insert(key, after);
+                    }
+                    Some(_) => return Err(InvalidConfigError::OldValueExists(key)),
```

Comment:
May be worthwhile to preserve what the old value was as well, so that whoever is editing the configuration doesn't need to go spelunking through config files looking for it.